tools/clang/test/CodeGenSPIRV/method.byte-address-buffer.templated-load.matrix.hlsl

```diff
@@ -33,12 +32,14 @@ void main(uint3 tid : SV_DispatchThreadId)
 // CHECK:         [[word2:%\d+]] = OpLoad %uint [[ptr]]
 // CHECK: [[shifted_word2:%\d+]] = OpShiftRightLogical %uint [[word2]] %uint_16
 // CHECK:          [[val5:%\d+]] = OpUConvert %ushort [[shifted_word2:%\d+]]
+// CHECK:          [[row0:%\d+]] = OpCompositeConstruct %v3ushort [[val0]] [[val1]] [[val2]]
 // CHECK:          [[row1:%\d+]] = OpCompositeConstruct %v3ushort [[val3]] [[val4]] [[val5]]
 // CHECK:        [[matrix:%\d+]] = OpCompositeConstruct %_arr_v3ushort_uint_2 [[row0]] [[row1]]
 // CHECK:                          OpStore %u16 [[matrix]]
-  uint16_t2x3 u16 = buf.Load<uint16_t2x3>(tid.x);
+  uint16_t2x3 u16 = buf.Load<column_major uint16_t2x3>(tid.x);
```

Comment:
Could you duplicate this test with these changes rather than modifying the existing one (to help verify that the default behavior hasn't changed)?

Reply:
I wanted to decouple the testing of defaults and matrix orders. In this test, I set everything explicitly so that we can check that we can handle various matrix shapes. In the order two new test, I tested that both defaults and overrides work as expected.